src/App.js

```diff
@@ -26,33 +27,123 @@ const generateSquares = () => {
 }
 
 const App = () => {
-
+  
   const [squares, setSquares] = useState(generateSquares());
+  const [turn, setTurn] = useState(true);
+  const [winnerExist, setWinner] = useState(false);
+  const [winner,showWinner] = useState("Let's play!");
 
-  // Wave 2
-  // You will need to create a method to change the square 
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
 
+  const changeSquare = (changedSquare)=>{
+      if(winnerExist === false){
+        const squareNew =[];
+        squares.forEach((row)=>{
+          const rowNew = [];
+          row.forEach((square)=>{
+            if(square.id===changedSquare.id && turn===true && changedSquare.value ===""){
+              changedSquare.value = PLAYER_1;
+              rowNew.push(changedSquare);
+            }
+            else if(square.id===changedSquare.id && turn===false && changedSquare.value ===""){
+              changedSquare.value = PLAYER_2;
+              rowNew.push(changedSquare);
+            }
+            else {
+              rowNew.push(square);
+            };
+          })
+          squareNew.push(rowNew);
+        })
+  
+        setSquares(squareNew);
+        checkForWinner(squareNew);
+        setTurn(!turn);
+        
+       
+      }
+      else {
+        console.log("winner is", winnerExist);
+      }
 
-  const checkForWinner = () => {
-    // Complete in Wave 3
 
   }
 
+  
+  
+
+  const checkForWinner = (squares) => {
+
+    if(squares[0][0].value===squares[0][1].value && squares[0][1].value===squares[0][2].value && squares[0][0].value !==""){
+      setWinner(true);
+      showWinner("The winner is " + squares[0][0].value);
+
+    }
+    else if 
+    (squares[1][0].value===squares[1][1].value && squares[1][1].value===squares[1][2].value && squares[1][0].value !==""){
+      setWinner(true);
+      showWinner("The winner is " + squares[1][0].value);
+    }
+    else if 
+    (squares[2][0].value===squares[2][1].value && squares[2][1].value===squares[2][2].value && squares[2][0].value !==""){
+      setWinner(true);
+      showWinner("The winner is " + squares[2][0].value);
+    }
+    else if 
+    (squares[0][0].value===squares[1][0].value && squares[1][0].value===squares[2][0].value && squares[0][0].value !==""){
+      setWinner(true);
+      showWinner("The winner is " + squares[0][0].value);
+    }
```

Comment:
This certainly works but could be "DRYed" up by setting this up in a loop format.